src/preact/base-element.js

```diff
@@ -725,6 +728,21 @@ function collectProps(Ctor, element, ref, defaultProps) {
   return props;
 }
 
+/**
+ * Clones an Element into a VNode.
+ * @param {!Element} element
+ * @return {!PreactDef.Renderable}
+ */
+function shallowCloneVNode(element) {
+  const {attributes} = element;
+  const props = {key: element};
+  for (let i = 0; i < attributes.length; i++) {
+    const {name, value} = attributes[i];
+    props[name] = value;
+  }
+  return Preact.createElement(element.tagName, props);
```

Comment:
IMHO we should do `toLowerCase()` on the tag name.

Reply:
Or use `localName`

Reply:
`createElement`'s definition is that if it's the type value is string, it's an element. Otherwise it's a component render reference.

But I get the defense on the framework potentially rejecting all-caps. I don't know if it's worth to keep the tiny performance benefit of not transforming the string. Changed.

Reply:
TIL about localName. Done.